packages/react-reconciler/src/ReactFiberWorkLoop.js

```diff
@@ -3341,23 +3367,18 @@ function commitRoot(
       ReactSharedInternals.T = prevTransition;
     }
   }
-  flushMutationEffects(root, finishedWork, lanes);
-  flushLayoutEffects(
-    root,
-    finishedWork,
-    lanes,
-    recoverableErrors,
-    didIncludeRenderPhaseUpdate,
-    suspendedCommitReason,
-    completedRenderEndTime,
-  );
+  pendingEffectsStatus = PENDING_MUTATION_PHASE;
```

Comment:
Is it a problem that we potentially leave `pendingEffectsRoot` as non-nullabble but `pendingEffectsStatus === NO_PENDING_EFFECTS` when we throw in `commitBeforeMutationEffects`?

Reply:
No, not really because it'll just leak until we update anything again.

A bigger problem would be leaving `pendingEffectsStatus` as something else. Leaving `PENDING_PASSIVE_PHASE` doesn't really happen because it's immediately reset and even if it did it would try again. Leaving `PENDING_MUTATION_PHASE` or `PENDING_LAYOUT_PHASE` would stop any new update except sync which would be bad.

So I'll reset it in the beginning in case React itself throws so it doesn't get stuck.